codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -282,7 +282,7 @@ private int allocateBytesForTree(Http2Stream parent, int connectionWindow) {
                 int weight = child.weight();
                 double weightRatio = weight / (double) totalWeight;
 
-                int bytesForTree = Math.min(nextConnectionWindow, (int) Math.ceil(connectionWindow * weightRatio));
+                int bytesForTree = Math.min(nextConnectionWindow, Math.max(1, (int) (connectionWindow * weightRatio)));
```

Comment:
Is there a possibility to mis-allocating resources with out the priority sort (by 1)?  Say if `nextConnectionWindow` is 1 we could end up allocating 1 to a lower weighted subtree and 0 to a higher weighted subtree (if the lower weighted subtree is before the higher weighted subtree).  Is this acceptable...does this violate the spec?